node/v2/handler.js

```diff
@@ -857,10 +849,49 @@ TChannelV2Handler.prototype.buildInRequest = function buildInRequest(reqFrame) {
         self.tracer
     );
 
+    var req;
     if (reqFrame.body.flags & v2.CallFlags.Fragment) {
-        return new StreamingInRequest(reqFrame.id, opts);
+        req = new StreamingInRequest(reqFrame.id, opts);
     } else {
-        return new InRequest(reqFrame.id, opts);
+        req = new InRequest(reqFrame.id, opts);
+    }
+
+    req.errorEvent.on(onReqError);
+
+    function onReqError(err) {
+        self.onReqError(err, req);
+    }
+
+    return req;
+};
+
+TChannelV2Handler.prototype.onReqError = function onReqError(err, req) {
+    var self = this;
+
+    var codeName = errors.classify(err);
+    if (codeName &&
+        codeName !== 'ProtocolError'
+    ) {
+        // TODO: move req to error state?
+        req.res = self.buildOutResponse(req);
```

Comment:
I think this entire code path is an "anti pattern".

We should favor `res.sendError()` which will correctly manage the entire OutResponse state machine.